services/src/test/scala/cromwell/services/ServicesStoreSpec.scala

```diff
@@ -93,6 +96,94 @@ class ServicesStoreSpec extends FlatSpec with Matchers with ScalaFutures with St
+          workflowOptions = None,
+          workflowState = "Submitted",
+          submissionTime = OffsetDateTime.now().toSystemTimestamp,
+          importsZip = None,
+          customLabels = "{}".toClob(refineMV[NonEmpty]("{}")),
+          cromwellId = None,
+          heartbeatTimestamp = Option(OffsetDateTime.now().toSystemTimestamp)
+        )
+      }
+      database.addWorkflowStoreEntries(entries).futureValue
+
+      def runDbFuture(futureNumber: Int): Future[Unit] = {
+        for {
+          // Alternates between running the pair sql statements listed in the deadlock report here:
+          // https://docs.google.com/document/d/1kNFz7CtGZ0BCkPP0XdwyQu3Bg1PKXR9qxZa4rkC27Js/edit
+          _ <- futureNumber % 2 match {
+            case 0 =>
+              logger.info(s"Selecting heartbeat-deadlock-test future #$futureNumber")
+              // select `WORKFLOW_EXECUTION_UUID`, `WORKFLOW_DEFINITION`, `WORKFLOW_ROOT`, `WORKFLOW_TYPE`, `WORKFLOW_TYPE_VERSION`, `WORKFLOW_INPUTS`, `WORKFLOW_OPTIONS`, `WORKFLOW_STATE`, `SUBMISSION_TIME`, `IMPORTS_ZIP`, `CUSTOM_LABELS`, `CROMWELL_ID`, `HEARTBEAT_TIMESTAMP`, `WORKFLOW_STORE_ENTRY_ID` from `WORKFLOW_STORE_ENTRY` where ((`HEARTBEAT_TIMESTAMP` is null) or (`HEARTBEAT_TIMESTAMP` < '2018-06-08 15:41:18.43')) and (not (`WORKFLOW_STATE` = 'On Hold')) order by `SUBMISSION_TIME` limit 50 for update
+              // select `WORKFLOW_EXECUTION_UUID`, `WORKFLOW_DEFINITION`, `WORKFLOW_ROOT`, `WORKFLOW_TYPE`, `WORKFLOW_TYPE_VERSION`, `WORKFLOW_INPUTS`, `WORKFLOW_OPTIONS`, `WORKFLOW_STATE`, `SUBMISSION_TIME`, `IMPORTS_ZIP`, `CUSTOM_LABELS`, `CROMWELL_ID`, `HEARTBEAT_TIMESTAMP`, `WORKFLOW_STORE_ENTRY_ID` from `WORKFLOW_STORE_ENTRY` where ((`HEARTBEAT_TIMESTAMP` is null) or (`HEARTBEAT_TIMESTAMP` < '2018-06-08 20:07:18.16')) and (not (`WORKFLOW_STATE` = 'On Hold')) order by `SUBMISSION_TIME` limit 50 for update
+              database.fetchWorkflowsInState(
+                limit = 50,
+                cromwellId = "heartbeat_deadlock_test",
+                heartbeatTimestampTimedOut = 30.seconds.ago,
+                heartbeatTimestampTo = OffsetDateTime.now.toSystemTimestamp,
+                workflowStateFrom = "Submitted",
+                workflowStateTo = "Running",
+                workflowStateExcluded = "On Hold"
+              )
+            case _ =>
+              logger.info(s"Updating heartbeat-deadlock-test future #$futureNumber")
+              // update `WORKFLOW_STORE_ENTRY` set `HEARTBEAT_TIMESTAMP` = '2018-06-08 15:51:17.661' where `WORKFLOW_STORE_ENTRY`.`WORKFLOW_EXECUTION_UUID` = '856a5363-4d6e-4c84-aafa-4cc94fb9c75b'
+              // update `WORKFLOW_STORE_ENTRY` set `HEARTBEAT_TIMESTAMP` = '2018-06-08 20:17:17.661' where `WORKFLOW_STORE_ENTRY`.`WORKFLOW_EXECUTION_UUID` = '4216b0cc-3b5e-4751-a3f5-24bfa5f7139a'
+              database.writeWorkflowHeartbeats(
+                workflowExecutionUuids = entries.take(10).map(_.workflowExecutionUuid).toSet,
```

Comment:
@kshakir @mcovarr I think what we were saying at retro was does this `entries.sortBy(_. submissionTime).take(10).map(_.workflowExecutionUuid).toSet` make the deadlock go away ?